### PR TITLE
Vendor `windows-sys` crate

### PR DESCRIPTION
**Pull request by NobodyXu**
to avoid additional dependencies bring in by `windows_sys`, espeically to `build-dependencies`.

### CODE REVIEW

**Review by thomcc — APPROVED** [review]
This is great, thank you.

**Comment by thomcc** [comment]
I guess `const fn transmute` is 1.56? I don't care that much about MSRV so I'd be okay bumping it. That's still very conservative anyway.

File: gen-windows-sys-binding/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+const PRELUDE: &str = r#"// This file is autogenerated.
+//
+// To add bindings, edit windows_sys.lst then run:
+//
+// ```
+// cd generate-windows-sys/
+// cargo run
+// ```
+//
+// ignore-tidy-filelength
```
[the remark this answers is not in the document]
Comment (NobodyXu) [reply gen-windows-sys-binding/src/main.rs]: Thanks, I've remove this.

**Comment by NobodyXu** [comment]
> I guess `const fn transmute` is 1.56? I don't care that much about MSRV so I'd be okay bumping it. That's still very conservative anyway.

I've replaced use of it with `addr as *mut T`, it will still work just a bit more derived from current std implementation, though it doesn't make much of a difference until std implements `strict_provenance`.

File: gen-windows-sys-binding/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+// ```
+// cd generate-windows-sys/
+// cargo run
+// ```
+"#;
+
+const POSTLUDE: &str = r#"
+/// Adapted from https://doc.rust-lang.org/src/core/ptr/mod.rs.html#600-607
+const fn invalid_mut<T>(addr: usize) -> *mut T {
+    addr as *mut T
```
Comment (thomcc) [inline gen-windows-sys-binding/src/main.rs:23]: A comment that this would be `transmute()` except for MSRV requirement might be good.

**Comment by NobodyXu** [comment]
> It's not really vendoring so much as using their windows-bindgen tool...

Well, anyway we are able to replace the massive `windows-sys` with 188 lines of rust.

File: gen-windows-sys-binding/src/main.rs (hunk 2 above)
Comment (NobodyXu) [reply gen-windows-sys-binding/src/main.rs:33]: Thanks!
I've added comment about MSRV and the strict_provenance.

**Comment by thomcc** [comment]
This looks great. Thanks for all your work.

**Pull request merged**